helix-view/src/view.rs

```diff
@@ -161,6 +161,62 @@ impl View {
         Some(Position::new(row, col))
     }
 
+    /// Verifies whether a screen position is inside the view
+    /// Returns true when position is inside the view
+    pub fn verify_screen_coords(&self, row: usize, column: usize) -> bool {
+        // 2 for status
+        if row < self.area.y as usize || row > self.area.y as usize + self.area.height as usize - 2
+        {
+            return false;
+        }
+
+        // TODO: not ideal
+        const OFFSET: usize = 7; // 1 diagnostic + 5 linenr + 1 gutter
+
+        if column < self.area.x as usize + OFFSET
+            || column > self.area.x as usize + self.area.width as usize
+        {
+            return false;
+        }
+        true
+    }
+
+    /// Translates a screen position to position in the text document.
+    /// Returns a usize typed position in bounds of the text if found in this view, None if out of view.
+    pub fn pos_at_screen_coords(&self, doc: &Document, row: usize, column: usize) -> Option<usize> {
+        if !self.verify_screen_coords(row, column) {
+            return None;
+        }
+
+        let text = doc.text();
+        let line_number = row - self.area.y as usize + self.first_line;
+        let mut pos = text.line_to_char(line_number);
+
+        let current_line = text.line(line_number);
+        let tab_width = doc.tab_width();
+
+        // TODO: not ideal
+        const OFFSET: usize = 7; // 1 diagnostic + 5 linenr + 1 gutter
+
+        let target = column - OFFSET - self.area.x as usize + self.first_col;
+        let mut selected = 0;
+
+        for grapheme in RopeGraphemes::new(current_line) {
+            if selected > target {
```

Comment:
This can be `>=`, and then we don't need the `- 1` on line 217 below.

Reply:
Yes. Both were changed because full line length (click after line end) needs to be reduced by 1.

Reply:
Yeah, both approaches will work.  I think the code ends up a little clearer with `>=` and handling the line ending explicitly, but I don't feel too strongly about it.

Having said that, the `Some(pos - 1)` at the end is still incorrect in your version.  It needs to work in terms of graphemes, so it should be `Some(prev_grapheme_boundary(text, pos))`.

(Your version will also break on the last line of the file after #376 is merged, since there's no guaranteed line break on the last line with that PR.  So we'll need to change it after that anyway.  But I don't think we need to block this PR for that.)